### PR TITLE
refactor: remove repetitive sanitization of post variables #3327

### LINKED ISSUE #3857: feat: create some general fn to handle super global variables
# Feature Request

## User Story
As a developer, I want to use super global variables properly so that their usage is sanitized properly and can help get rid of WPCS errors to achieve better score when Tide Project is implemented sooner or later.

Sometimes, there are code found in core or add-ons where we have not sanitized the super global variables like `$_GET`, `$_POST`, and `$_SERVER` which can lead to possible vulnerabilities.

## Possible Solution
Create a general function (maybe name it `give_get_global_variable`) which will contain a parameter `$type` (GET, POST, and SERVER) then in that function we will wrap the superglobal variables with `filter_input_array` method of PHP and `give_clean` method of Give core.

**Note:** We will create the function for now and will test whether it is working fine or not. Then, we will use it whenever we will find it in direct access usage of super global variables in Give Core.

## Acceptance Criteria
- [x] Create a general function
- [x] Test the function
- [x] Do not cache sanitize super global because the code will break if any add-on or core changes super global manually. https://github.com/impress-org/give/blob/b10f3fabec43b2ffd2b0e78f6149c52db1582849/includes/process-donation.php#L185

### PR DESCRIPTION
**Pull request by ravinderk**
## Description
This PR will resolve #3857 #3327 

**Note: I created helper function as per https://github.com/impress-org/give/issues/3857 and refactor code as we described here https://github.com/impress-org/give/issues/3327. I create a sperate issue to refactor whole core code:https://github.com/impress-org/give/issues/4103**

## How Has This Been Tested?
Process donation and PHPUnit tests.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/formatting.php [hunk 1]
```diff
@@ -674,6 +674,51 @@ function give_get_cache_key( $action, $query_args ) {
...
+			break;
+
+		case 'GET':
+			$result = give_clean( $_GET );  // WPCS: input var ok, sanitization ok, CSRF ok.
+			break;
+	}
+
+	$give_super_global[ $cache_key ] = $result;
+
+	return $result;
```
Comment (DevinWalker) [inline includes/formatting.php]: Filter here?

File: includes/formatting.php [hunk 2]
```diff
@@ -674,6 +674,51 @@ function give_get_cache_key( $action, $query_args ) {
...
+/**
+ * Get sanitized super global param
+ *
+ * @param string $type
+ *
+ * @return mixed
+ * @since 2.5.0
+ *
+ */
+function give_get_super_global( $type ) {
```
Comment (DevinWalker) [inline includes/formatting.php]: What about a second param to get an array key from the super global like `$_POST['give-gateway']`?

File: includes/formatting.php (hunk 1 above)
Comment (ravinderk) [reply includes/formatting.php]: @DevinWalker Check updated code. It is moved somewhere else: https://github.com/impress-org/give/pull/4102/commits/f54fac38570f99b60c5384a2699276503cd36922

File: includes/formatting.php (hunk 2 above)
Comment (ravinderk) [reply includes/formatting.php]: @DevinWalker I think, this is a good suggestion. I will update that.

**Pull request merged**